### PR TITLE
Add kmini

### PR DESCRIPTION
**Pull request by Maartenwut**
Add support for the kmini. The keyboard only has one layout afaik, correct me if I'm wrong.

### CODE REVIEW

File: keyboards/kmini/readme.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Kmini keyboard firmware
+======================
+
+The Revo Kmini is a 65% keyboard with an additional two columns on the left. It is programmed with the 'KMAC key map', which only works on Windows and according to a [user](https://www.keebtalk.com/t/programming-revo-kmini/2107/7) it cannot map media controls or change the capslock key.
```
Comment (drashna) [inline keyboards/kmini/readme.md]: Feel free to use my username here. ;)

And to clarify, you can change the capslock.  I'm not sure how the reset works on it, though (if it's looking at the keycode, or if it's looking at the matrix location),  

Also, I didn't see the media keys, so I'm not sure about those

Comment (Maartenwut) [reply keyboards/kmini/readme.md]: I ported this for a friend of mine, he claimed that he wasn't able to change the capslock key, even though he set it to control. Anyways, the reset function is hardwired and should not depend on the keymap. I don't know if you ported the kmac, but if you did, thanks! You helped a lot with this port 👍

**Comment by drashna** [comment]
Thanks!

File: keyboards/kmini/readme.md (hunk 1 above)
Comment (drashna) [reply keyboards/kmini/readme.md]: Not a problem.  And no it wasn't me.

THough, looks like there is an issue with the board though, at least for the QMK configurator. 

And good to know, on the rest

File: keyboards/kmini/matrix.c [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+
+/* matrix state(1:on, 0:off) */
+static matrix_row_t matrix[MATRIX_ROWS];
+static matrix_row_t matrix_debouncing[MATRIX_ROWS];
+
+static void init_rows(void);
+static bool read_rows_on_col(matrix_row_t current_matrix[], uint8_t current_col);
+static void unselect_cols(void);
+static void select_col(uint8_t col);
+
```
Comment (drashna) [inline keyboards/kmini/matrix.c]: To fix this on config.qmk.fm, we need to add: 
```c
__attribute__ ((weak))
void matrix_init_user(void) {}

__attribute__ ((weak))
void matrix_scan_user(void) {}

__attribute__ ((weak))
void matrix_init_kb(void) {
  matrix_init_user();
}

__attribute__ ((weak))
void matrix_scan_kb(void) {
  matrix_scan_user();
}
```

If you wanted to make this change, and open a new PR, that should fix the compiling issue

**Pull request merged**